internal/checkers/compares.go

```diff
@@ -42,13 +46,24 @@ func (checker Compares) Check(pass *analysis.Pass, call *CallMeta) *analysis.Dia
 		return nil
 	}
 
+	_, ptrComparison := pass.TypesInfo.TypeOf(be.X).(*types.Pointer)
+
 	var tokenToProposedFn map[token.Token]string
 
 	switch call.Fn.NameFTrimmed {
```

Comment:
This code works, but then it becomes very difficult to read. I mean a switch with 2 if in it.

Maybe code could be refactored to use a function to return the token suggestion.

Or the map, could have extra level.

Just a suggestion, you can of course discard

Reply:
> Or the map, could have extra level.

I thought about it, but did not find the appropriate extendable key

But I simplified the code, thanks 👌

Reply:
I confirm, it's way better now. The code is good and clear. I don't think the map thing would have been as simple as your implementation. So I'm very happy with it.